### ISSUE TITLE
Faulty generated keys

### ISSUE THREAD

**Issue by spotlightishere** [OpenShopChannel/WSC-Patcher#2]
Based on [a comment in WSC-Web#10](https://github.com/OpenShopChannel/WSC-Web/discussions/10#discussioncomment-2302169), keys generated via our own store provide this error:

```
ssl.SSLError: [X509: KEY_VALUES_MISMATCH] key values mismatch (_ssl.c:4065)
```

Needs investigation.

While making SSL-related changes, determining if 2048-bit keys across the board are suitable would be preferable.

**Comment by spotlightishere** [comment]
As of 952c3824e3cf39ebfac2f5a860661e17c53aae42, this should be resolved.

I took the opportunity to bump the CA certificate's size to 2048 bits. Keeping this issue open to more throughly test on a console, as IOS (as requested via EC) differed from Opera per testing.

Please let me know if you can verify this functions on a console with EC set up!

**Comment by TheTechRobo** [comment]
Can confirm that this is fixed in the new branch. Can't test EC, though.

**Comment by spotlightishere** [comment]
I was able to test this branch on a physical Wii - while Opera and Dolphin HLE (against Mbed TLS) are okay with the certificate as generated, oddly IOS (as used with EC) was not, rejecting it as an unknown certificate.


I was able to get IOS to load the generated certificate (retaining 2048-bit CA!) in 4a232c552b39cd597f7c6a3a90bdda6f6d49072e. It seems specifying a SAN _and_ constraints allows it to accept things properly. Quite honestly, I have no clue why, but.. it works 😅
